Content.Server/GameObjects/Components/Movement/PlayerInputMoverComponent.cs

```diff
@@ -108,8 +128,11 @@ public void SetVelocityDirection(Direction direction, bool enabled)
             x += _movingRight ? 1 : 0;
 
             var y = 0;
-            y -= _movingDown ? 1 : 0;
-            y += _movingUp ? 1 : 0;
+            if (DiagonalMovementEnabled || (!_movingLeft && !_movingRight))
```

Comment:
if (DiagonalMovementEnabled || x == 0)

this will help ya if other stuff changes x velocity

Reply:
Addressed